src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -104,7 +114,48 @@ protected function permissions_check() {
 	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
 	 */
 	public function get_item_permissions_check( $request ) {
-		return $this->permissions_check( $request );
+		$post = $this->get_post( $request['id'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		if ( 'edit' === $request['context'] && $post && ! $this->check_update_permission( $post ) ) {
+			return new WP_Error(
+				'rest_forbidden_context',
+				__( 'Sorry, you are not allowed to edit this global style.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( $post ) {
+			return $this->check_read_permission( $post );
+		}
+
+		return $this->permissions_check();
+	}
+
+	/**
+	 * Checks if a global style can be read.
+	 *
+	 * Correctly handles posts with the inherit status.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param WP_Post $post Post object.
+	 * @return bool Whether the post can be read.
+	 */
+	public function check_read_permission( $post ) {
+		// Is the post readable?
+		if ( 'publish' === $post->post_status || current_user_can( 'read_post', $post->ID ) ) {
+			return true;
+		}
+
+		$post_status_obj = get_post_status_object( $post->post_status );
+		if ( $post_status_obj && $post_status_obj->public ) {
+			return true;
+		}
+
+		return false;
```

Comment:
```suggestion
		return current_user_can( 'read_post', $post->ID );
```

`trunk` doesn't allow logged out users to view the custom styles so the public/private nature of the post status shouldn't be considered.

If your thinking is to allow developers to filter things so it can, then `is_post_publicly_viewable( $post )` is a combined check of post the post type's and status's publicly viewable status.

Reply:
Fixed.